tests/robustness/snapshotter.go

```diff
@@ -3,13 +3,26 @@
 // can be used to test other environments.
 package robustness
 
+import (
+	"io"
+	"time"
+)
+
 // Snapshotter is an interface that describes methods
 // for taking, restoring, deleting snapshots, and
 // tracking them by a string snapshot ID.
 type Snapshotter interface {
-	CreateSnapshot(sourceDir string, opts map[string]string) (snapID string, err error)
-	RestoreSnapshot(snapID string, restoreDir string, opts map[string]string) error
 	DeleteSnapshot(snapID string, opts map[string]string) error
 	RunGC(opts map[string]string) error
 	ListSnapshots() ([]string, error)
+
+	CreateSnapshot(sourceDir string, opts map[string]string) (snapID string, fingerprint []byte, stats *CreateSnapshotStats, err error)
+	RestoreSnapshot(snapID, restoreDir string, opts map[string]string) ([]byte, error)
+	RestoreSnapshotCompare(snapID, restoreDir string, validationData []byte, reportOut io.Writer, opts map[string]string) error
+}
+
+// CreateSnapshotStats is a struct for returning various stats from the snapshot execution.
+type CreateSnapshotStats struct {
+	SnapStartTime time.Time
+	SnapEndTime   time.Time
 }
```

Comment:
Interesting that you introduce this here. Could we add an arbitrary data field like `Raw []byte`.

I was looking for somewhere to track some CBT related things (if I can get them out of K10) like the #blocks, etc.

Reply:
[Added Raw field here](https://github.com/kastenhq/kopia/pull/125/commits/ea9e565d8892a46f83e6459c8f58deb2a4a8bd6a)